waku/waku_lightpush/client.nim

```diff
@@ -110,3 +114,34 @@ proc publishToAny*(
     obs.onMessagePublished(pubSubTopic, message)
 
   return lightpushSuccessResult(publishedCount)
+
+proc publishWithConn*(
```

Comment:
Nitpick and not for this PR: we can probably clean up Lightpush API a little bit by reducing the number of `publishX` methods and using `Option` more judiciously (e.g. `publish(LightpushRequest, Option[Connection])`)